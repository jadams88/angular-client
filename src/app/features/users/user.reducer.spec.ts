import { usersReducer } from '@features/users/user.reducer';
import { User } from '@features/users/user.model';
import {
  LoadUserSuccess,
  UpdateUserSuccess,
  DeleteUserSuccess
} from '@features/users/user.actions';

describe('UsersReducer', () => {
  describe('undefined action', () => {
    it('should return the default state', () => {
      const action = {} as any;
      const result = usersReducer(undefined, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('LoadSucces action', () => {
    it('should add the user to the user state', () => {
      const user: User = { id: '2', name: 'someUser', email: 'some@email.com' };
      const action = new LoadUserSuccess(user);

      const result = usersReducer(undefined, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('UpdateSucces action', () => {
    it('should update the existing user state', () => {
      const inititalState: User = { id: '1', name: 'initialUser', email: 'some@email.com' };
      const user = { id: '1', name: 'updatedName' } as User;
      const action = new UpdateUserSuccess(user);

      const result = usersReducer(inititalState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('DeleteSucces action', () => {
    it('should delete the existing user state', () => {
      const inititalState: User = { id: '1', name: 'initialUser', email: 'some@email.com' };
      const action = new DeleteUserSuccess();

      const result = usersReducer(inititalState, action);

      expect(result).toMatchSnapshot();
    });
  });
});
