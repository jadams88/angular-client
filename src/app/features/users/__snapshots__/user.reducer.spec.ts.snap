// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersReducer DeleteSucces action should delete the existing user state 1`] = `undefined`;

exports[`UsersReducer LoadSucces action should add the user to the user state 1`] = `
Object {
  "email": "some@email.com",
  "id": "2",
  "username": "someUser",
}
`;

exports[`UsersReducer UpdateSucces action should update the existing user state 1`] = `
Object {
  "email": "some@email.com",
  "id": "1",
  "username": "updatedName",
}
`;

exports[`UsersReducer undefined action should return the default state 1`] = `undefined`;
